tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionSetCreatePerformer.scala

```diff
@@ -0,0 +1,86 @@
+package com.linagora.tmail.james.jmap.method
+
+import com.linagora.tmail.james.jmap.firebase.FirebaseSubscriptionRepository
+import com.linagora.tmail.james.jmap.json.FirebaseSubscriptionSerializer
+import com.linagora.tmail.james.jmap.method.FirebaseSubscriptionSetCreatePerformer.{CreationFailure, CreationResult, CreationResults, CreationSuccess}
+import com.linagora.tmail.james.jmap.model.{DeviceClientIdInvalidException, ExpireTimeInvalidException, FirebaseSubscriptionCreation, FirebaseSubscriptionCreationId, FirebaseSubscriptionCreationParseException, FirebaseSubscriptionCreationRequest, FirebaseSubscriptionCreationResponse, FirebaseSubscriptionExpiredTime, FirebaseSubscriptionSetRequest}
+import eu.timepit.refined.auto._
+import org.apache.james.core.Username
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.core.{Properties, SetError}
+import org.slf4j.{Logger, LoggerFactory}
+import play.api.libs.json.JsObject
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import javax.inject.Inject
+
+
+object FirebaseSubscriptionSetCreatePerformer {
+  private val LOGGER: Logger = LoggerFactory.getLogger(classOf[FirebaseSubscriptionSetCreatePerformer])
+
+  trait CreationResult
+
+  case class CreationSuccess(clientId: FirebaseSubscriptionCreationId, response: FirebaseSubscriptionCreationResponse) extends CreationResult
+
+  case class CreationFailure(clientId: FirebaseSubscriptionCreationId, e: Throwable) extends CreationResult {
+    def asMessageSetError: SetError = e match {
+      case e: FirebaseSubscriptionCreationParseException => e.setError
+      case e: ExpireTimeInvalidException => SetError.invalidArguments(SetErrorDescription(e.getMessage), Some(Properties("expires")))
+      case e: DeviceClientIdInvalidException => SetError.invalidArguments(SetErrorDescription(e.getMessage), Some(Properties("deviceClientId")))
+      case e: IllegalArgumentException => SetError.invalidArguments(SetErrorDescription(e.getMessage))
+      case _ => {
```

Comment:
Please handle also `TokenInvalidException` (duplicate FCM token) and a test for it.